enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go

```diff
@@ -0,0 +1,266 @@
+		Contents: args.Input.FileContents,
+		Proto:    proto,
+	}
+	if err := r.codeownersStore.UpdateCodeownersFile(ctx, codeownersFile); err != nil {
+		return nil, errors.Wrap(err, "could not update codeowners file")
+	}
+
+	return &codeownersIngestedFileResolver{
+		codeownersFile: codeownersFile,
+		repository:     repo,
+		db:             r.db,
+		gitserver:      r.gitserver,
+	}, nil
+}
+
+func parseInputString(fileContents string) (*codeownerspb.File, error) {
+	fileReader := strings.NewReader(fileContents)
+	ruleset, err := codeowners.Parse(fileReader)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not parse input")
+	}
+	return ruleset.GetFile(), nil
+}
+
+func (r *ownResolver) getRepo(ctx context.Context, input graphqlbackend.CodeownersFileInput) (*types.Repo, error) {
+	if input.RepoID == nil && input.RepoName == nil {
+		return nil, errors.New("either RepoID or RepoName should be set")
+	}
+	if input.RepoID != nil && input.RepoName != nil {
+		return nil, errors.New("both RepoID and RepoName cannot be set")
+	}
+	if input.RepoName != nil {
+		repo, err := r.db.Repos().GetByName(ctx, api.RepoName(*input.RepoName))
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not fetch repository for name %v", input.RepoName)
+		}
+		return repo, nil
+	}
+	repoID, err := graphqlbackend.UnmarshalRepositoryID(*input.RepoID)
+	if err != nil {
+		return nil, err
+	}
+	return r.db.Repos().Get(ctx, repoID)
+}
+
+func (r *ownResolver) DeleteCodeownersFiles(ctx context.Context, args *graphqlbackend.DeleteCodeownersFileArgs) (*graphqlbackend.EmptyResponse, error) {
+	if err := r.viewerCanAdminister(ctx); err != nil {
+		return nil, err
+	}
+	if err := r.codeownersStore.DeleteCodeownersForRepos(ctx, args.RepositoryIDs...); err != nil {
+		return nil, errors.Wrapf(err, "could not delete codeowners file for repos +%d", args.RepositoryIDs)
+	}
+	return &graphqlbackend.EmptyResponse{}, nil
+}
+
+func (r *ownResolver) CodeownersIngestedFiles(ctx context.Context, args *graphqlbackend.CodeownersIngestedFilesArgs) (graphqlbackend.CodeownersIngestedFileConnectionResolver, error) {
+	if err := r.viewerCanAdminister(ctx); err != nil {
+		return nil, err
+	}
+	connectionResolver := &codeownersIngestedFileConnectionResolver{
+		codeownersStore: r.codeownersStore,
+	}
+	if args.After != nil {
+		cursor, err := graphqlutil.DecodeIntCursor(args.After)
+		if err != nil {
+			return nil, err
+		}
+		connectionResolver.cursor = int32(cursor)
+		if int(connectionResolver.cursor) != cursor {
+			return nil, errors.Newf("cursor int32 overflow: %d", cursor)
+		}
+	}
+	if args.First != nil {
+		connectionResolver.limit = int(*args.First)
+	}
+	return connectionResolver, nil
+}
+
+func (r *ownResolver) RepoIngestedCodeowners(ctx context.Context, repoID api.RepoID) (graphqlbackend.CodeownersIngestedFileResolver, error) {
+	// This endpoint is open to anyone.
+	// TODO: do we need to check repo permissions here, or is that check embedded in the repos store?
```

Comment:
It is, but good thinking! Maybe add a comment about that 

```suggestion
	// Repo store makes sure the viewer has access to the repository.
```